### ISSUE TITLE
opening moves for the 400 test matches should have some random play (or use some percentage of fixed opennings)

### ISSUE THREAD

**Issue by LetterRip** [gcp/leela-zero#506]
I think there is a serious chance of overfitting without any randomness for the openning in the test matches - I don't think I've seen any game that hasn't started out the same first couple of moves - and it risks overfitting with the player that memorizes the best openning sequence.  Perhaps there should be a standard suite of opennings should be used for some percentage of the opennings (say 50%).  This was we don't end up with overfitting as much and yet if a novel openning that out performs the typical opennings occurs it can be used.

**Comment by evanroberts85** [comment]
The training does not use match games as input data, so they can not lead to overfitting. AGZ did not use random openings for verification matches and we are following that.

**Comment by LetterRip** [comment]
"The training does not use match games as input data, so they can not lead to overfitting."

Are you familiar with "parameter tweak overfitting"

"Use a learning algorithm with many parameters. Choose the parameters based on the test set performance. For example, choosing the features so as to optimize test set performance can achieve this."

http://hunch.net/?p=22

We are passing information to our training set because we are retesting based on progress of beating the performance of the last successfully trained network.

**Comment by evanroberts85** [comment]
I had thought about that possibility, was not aware though of it being a
known effect. I do not think it applies here for Leela though.

**Comment by gcp** [comment]
We are following the AGZ paper here.

**Comment by gcp** [comment]
The optimization goal on the learning is different from the one in the validation, and there's randomness in the engine itself due to the assymetry of the net evaluations, as well as the networks being unequal.

There's no evidence nor even any indications this could be a problem.
